src/openapi.yaml

```diff
@@ -95,6 +74,62 @@ paths:
     get:
       operationId: rol_list
       description: Opvragen en bewerken van ROLrelatie tussen een ZAAK en een BETROKKENE.
+      parameters:
+        - name: zaak
+          in: query
+          description: URL to the related resource
+          required: false
+          schema:
+            type: string
+            format: uri
+        - name: betrokkene
+          in: query
+          description: ''
+          required: false
+          schema:
+            type: string
+            format: uri
+        - name: betrokkene_type
+          in: query
+          description: ''
+          required: false
+          schema:
+            type: string
+            enum:
+              - Natuurlijk persoon
+              - Niet-natuurlijk persoon
+              - Vestiging
+              - Organisatorische eenheid
+              - Medewerker
```

Comment:
Als ik het dus goed heb, wordt hier "polymorphisme" toegepast op `betrokkene` (url) waarbij het type resource wordt bepaald door `betrokkene_type`.

Hier wordt dan impliciet een beslissing genomen dat dit de manier is om dit soort scenario's aan te pakken. Denk dat we dit moeten documenteren als [design keuze](https://github.com/VNG-Realisatie/gemma-zaken/blob/master/docs/content/developers/design-keuzes.md)

Ik denk inderdaad dat dit de beste manier is maar het zou expliciet in de `description` moeten staan wat de relatie is tussen deze 2 velden, zoiets als:

* betrokkene: URL naar resource van type zoals gedefinieerd in `betrokkene_type`.
* betrokkene_type: Bepaald het type resource dat terugkomt via `betrokkene`.

Een implementatie detail is dan ook echt dat `betrokkene_type` WEL verplicht is als `betrokkene` is ingevuld. In OAS is het [nog niet mogelijk](https://github.com/OAI/OpenAPI-Specification/issues/256) om dit soort relaties aan te geven.

Reply:
Kan in plaats van een verplichte vulling (niet mogelijk in OAS) met een default waarde worden gewerkt of zo?

Reply:
Mijn laatste opmerking was ongeldig, zie edit in commentaar.